parquet/src/arrow/record_reader/mod.rs

```diff
@@ -202,6 +203,37 @@ where
         Ok(records_read)
     }
 
+    /// Try to skip the next `num_records` rows
+    ///
+    /// # Returns
+    ///
+    /// Number of records skipped
+    pub fn skip_records(&mut self, num_records: usize) -> Result<usize> {
+        // First need to clear the buffer
+        let (buffered_records, buffered_values) = self.count_records(num_records);
+        self.num_records += buffered_records;
+        self.num_values += buffered_values;
+
+        self.consume_def_levels();
+        self.consume_rep_levels();
+        self.consume_record_data();
+        self.consume_bitmap();
+        self.reset();
+
+        let remaining = buffered_records - num_records;
```

Comment:
Is there a situation, where `buffered_record`s less than `num_records`🤔

Reply:
both `buffered_records ` and `buffered_records` are usize it will overflow,
and if `remaining ` is nagtive how to use in ` column_reader.skip_records(remaining)?,`

Reply:
Oh sorry, I wrote that response on my phone and got my wires crossed. No buffered_records here <= num_records - count_records cannot return more than the `records_to_read` argument which in this case is `num_records`.

Reply:
I thinks it should be 🤔:
```suggestion
        let remaining = num_records - buffered_records;
```